src/main/java/Interactors/OutputInteractor.java

```diff
@@ -0,0 +1,122 @@
+package Interactors;
+
+import Entities.Output;
+import Entities.State;
+import UseCases.UseCaseInteractor;
+
+import java.util.ArrayList;
+
+/***
+ * OutputInteractor is a class that controls the data in the Output Entity and interacts with the UseCaseInteractor
+ * to present output the user based on the current state of the game
+ */
+public class OutputInteractor {
+    /**
+     *  InsatanceVar output: an Output object that keeps track and updates the output depending on the state of the game
+     *  InstanceVar currentState: a State object that helps with determining the output that should be presented to the user
+     */
+    private Output output;
+    private State currentState;
+
+    /**
+     * The Constructor for the OutputInteractor Class
+     * @param interactor: gets the initial state of the game
+     */
+    public OutputInteractor(UseCaseInteractor interactor){
+        this.output = new Output();
+        this.currentState = interactor.getInitialState();
+    }
+
+    /**
+     * setFinalOutput updates the string to be presented to the user by updating all the output strings that should
+     * be presented to the user based on the state and presented all the options based on the state
+     */
+    public void setFinalOutput(){
+        updateLogicStates(this.currentState.getId());
+        this.output.setFinalOutput(this.output.getStateOutput(this.currentState.getId()));
+        addOptionStrings();
+    }
+
+    /**
+     * This function deals with all the states that need to be updated periodically based on the current state of the game
+     * @param state: the current state that game is in
+     */
+    public void updateLogicStates(String state){
+        StringBuilder currString = new StringBuilder();
+        switch (state) {
+            case "MainTree":
```

Comment:
Should we use a switch statement here, isn't that a code smell? Also, the method is kinda long, which would be a code smell, more specifically a bloater, wouldn't it be better if we refactor these out to different methods and then after that just call those methods in this method?

Reply:
I agree with this feedback, it looks like the code follows the same logic for each case. Could you try to refactor the switch statement by creating a helper method following the logic in each case body and which takes as input the parameters which change?

Reply:
I address the comment about splitting this up, but with regards to the code smell about switch statements, that will be dealt with in a later refactor with regards to actually creating an abstract class for the different states and the child classes which all output their respective state in a string